python/tests/test_fdata.py

```diff
@@ -492,3 +507,29 @@ def test_fdata_dimension(tmpdir_factory, merge_files_manyLR):
         assert list(curves[0][4][1])    == [2]
         assert list(curves[0][5][0])    == [1]
         assert list(curves[0][6])       == [1, 2, 3, 4]
+
+def test_fdata_tuple_dimension():
+    fpath = 'data/chap4-7/iflr/multidimensions-validated.dlis'
+
+    with dlisio.load(fpath) as (f, *_):
+        frame = f.object('FRAME', 'FRAME-VALIDATE', 10, 0)
+        curves = frame.curves()
+
+        assert curves[0][0].size == 3
+
+        assert curves[0][0][0] == (56, 0.0625, 0.0625)
+        assert curves[0][0][1] == (43, 0.0625, 0.0625)
+        assert curves[0][0][2] == (71, 0.5, 0.5)
+
+@pytest.mark.xfail(strict=True)
+def test_fdata_dimensions_in_multifdata():
+    fpath = 'data/chap4-7/iflr/multidimensions-multifdata.dlis'
+
+    with dlisio.load(fpath) as (f, *_):
+        frame = f.object('FRAME', 'FRAME-DIMENSION', 11, 0)
+        curves = frame.curves()
+
+        assert list(curves[0][0][0]) == [1, 2, 3]
```

Comment:
Use numpy.testing.assert_array_equal [1], it produces a way cleaner and more informative output when failing

[1] https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_array_equal.html